packages/app-accounts/src/Creator.tsx

```diff
@@ -207,15 +207,25 @@ class Creator extends React.PureComponent<Props, State> {
   }
 
   onCommit = (): void => {
-    const { onBack } = this.props;
     const { name, password, seed } = this.state;
     const pair = keyring.createAccount(
       formatSeed(seed), password, { name }
     );
 
     InputAddress.setLastValue('account', pair.address());
 
-    onBack();
+    const json = pair.toJson(password);
+
+    // add encrypted data to keyring since immediately after account creation only secret key is in memory
+    keyring.loadAccount(json);
+
+    this.onCreate();
```

Comment:
We can actually just call directly to `onCreateAccount` here instead of having the member function that add nothing special, it just calls the function, nothing else.